end-to-end-tests/tst/test_http_api_cors.py

```diff
@@ -0,0 +1,47 @@
+import boto3
+import os
+from helpers.base_test import BaseTest
+from helpers.resources import Resources
+from helpers.resource_types import ResourceTypes
+
+
+class HttpApiCorsTest(BaseTest):
+    http_api_with_cors_resources = (
+        Resources()
+        .created("HttpApiFunction", ResourceTypes.LAMBDA_FUNCTION)
+        .created("HttpApiFunctionRole", ResourceTypes.IAM_ROLE)
+        .created("HttpApiFunctionImplicitApiPermission", ResourceTypes.LAMBDA_PERMISSION)
+        .created("ServerlessHttpApi", ResourceTypes.APIGWV2_HTTPAPI)
+        .created("ServerlessHttpApiApiGatewayDefaultStage", ResourceTypes.APIGWV2_STAGE)
+    )
+
+    def tearDown(self):
+        self.delete_stack()
+
+    def test_cors(self):
+        input_template = os.path.join(os.getcwd(), "end-to-end-tests/input", "http_api_cors", "template.yaml")
+        capabilities = ["CAPABILITY_IAM"]
+        stack_name = self.make_and_verify_stack(input_template, capabilities, self.http_api_with_cors_resources)
```

Comment:
Can we split the make and verify steps into separate functions? It will make it easier to reason about when each step does one operation.

Reply:
Agreed. I have added three functions: `make_stack`, `verify_stack` and `make_and_verify_stack`. We can either use make step then verify or use a single function `make_and_verify_stack` now